adobe_vipm/flows/fulfillment/purchase.py

```diff
@@ -89,44 +89,37 @@ class ValidateMarketSegmentEligibility(Step):
 
     def __call__(self, client, context, next_step):
         """Validate if the customer is eligible to place orders for a given market segment."""
-        if context.market_segment != MARKET_SEGMENT_COMMERCIAL:
-            if not self._validate_agency_type(client, context):
-                return
+        if context.market_segment == MARKET_SEGMENT_EDUCATION and not context.adobe_customer.get(
+            "companyProfile", {}
+        ).get("marketSubSegments"):
```

Comment:
I'd like to move this condition into a method to make it more  readable and easier toto understand